File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string ,vector<string>> mp;

        for(string x: strs){
            string temp= x;
            sort(temp.begin(), temp.end());
            mp[temp].push_back(x);
        } 

        vector<vector<string>> ans;

        for(auto [key, value]: mp){
            ans.push_back(value);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> mp = new HashMap<>();

        for (String x : strs) {
            String temp = x;
            char[] arr = temp.toCharArray();
            Arrays.sort(arr);
            temp = new String(arr);

            if (!mp.containsKey(temp)) {
                mp.put(temp, new ArrayList<>());
            }
            mp.get(temp).add(x);
        }

        List<List<String>> ans = new ArrayList<>();

        for (Map.Entry<String, List<String>> entry : mp.entrySet()) {
            ans.add(entry.getValue());
        }

        return ans;
    }
}
